compiler/noirc_frontend/src/parser/parser.rs

```diff
@@ -369,10 +370,41 @@ fn optional_type_annotation<'a>() -> impl NoirParser<UnresolvedType> + 'a {
         .map(|r#type| r#type.unwrap_or_else(UnresolvedType::unspecified))
 }
 
+/// import_visibility: 'pub(crate)'? 'pub'? ''
+fn import_visibility() -> impl NoirParser<ItemVisibility> {
+    let is_pub_crate = (keyword(Keyword::Pub)
+        .then_ignore(just(Token::LeftParen))
+        .then_ignore(keyword(Keyword::Crate))
+        .then_ignore(just(Token::RightParen)))
+    .map(|_| ItemVisibility::PublicCrate);
+
+    let is_pub_super = (keyword(Keyword::Pub)
+        .then_ignore(just(Token::LeftParen))
+        .then_ignore(keyword(Keyword::Super))
+        .then_ignore(just(Token::RightParen)))
+    .map(|_| ItemVisibility::PublicSuper);
+
+    let is_pub = keyword(Keyword::Pub).map(|_| ItemVisibility::Public);
+
+    let is_private = empty().map(|_| ItemVisibility::Private);
+
+    choice((is_pub_crate, is_pub_super, is_pub, is_private))
+}
```

Comment:
This is a bit confusing as we're overloading `pub` for both zk visibiilty and standard programming language visibility.

An alternative would be to rename this to `export` so we'd have `export(super)` and `export(crate)` to avoid this overloading.

Reply:
It's a break from rust but a relatively weak one as it's just a keyword change.

Reply:
> a break from rust
`pub` for ZK is a break anyway; the benefits of more intuitive keywords almost always outweigh the costs.

Reply:
@jfecher what are your thoughts on this?

Reply:
It sounds like using `pub` is a compromise either way. `pub` for item visibility is much more common within Noir while `pub` for zk is only used on entry-point function parameters. Changing either would be breaking but changing `pub` visibility may be less so just because it is newer. Using `export fn`, `export(crate) fn`, etc would make Noir less rust-like which could be a stumbling block. I'm fine with both uses of `pub` for now but if there's ever a future scenario where we could want to apply `pub (zk)` to a function somehow then things would get confusing real fast.